Application/Frontend/pages/api/get-message-history.tsx

```diff
@@ -0,0 +1,50 @@
+import type { NextApiRequest, NextApiResponse } from 'next';
```

Comment:
Overall, great job of effectively handling fetching message history from a MongoDB database through a RESTful API endpoint, ensuring proper error handling and response generation for various scenarios.